test/functional/rpc_rawtransaction.py

```diff
@@ -62,44 +62,51 @@ def set_test_params(self):
         # whitelist all peers to speed up tx relay / mempool sync
         for args in self.extra_args:
             args.append("-whitelist=noban@127.0.0.1")
+        if self.is_wallet_compiled():
+            self.requires_wallet = True
 
         self.supports_cli = False
 
     def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
+        if self.is_wallet_compiled():
+            if self.options.descriptors:
+                self.skip_if_no_sqlite()
+            else:
+                self.skip_if_no_bdb()
```

Comment:
Interesting. I think the goal of this change it to not run the test at all to avoid wasting resources if a specific wallet type is requested, but not available?

However, the test will still run twice when the wallet isn't compiled: https://cirrus-ci.com/task/5291414786408448?logs=ci#L3574

I think it might be better to fix this by removing this whole method and also consolidate the two tasks into one:

```
$ git grep 'rpc_rawtransaction.py'
test/functional/test_runner.py:    'rpc_rawtransaction.py --legacy-wallet',
test/functional/test_runner.py:    'rpc_rawtransaction.py --descriptors',

Reply:
> I think the goal of this change it to not run the test at all to avoid wasting resources if a specific wallet type is requested, but not available?

It's more that without this function if you try to run the test with `--descriptors` but without having sqlite compiled (or, viceversa, `legacy` but no bdb) the whole test fails... 

```
❯ ./configure --without-bdb
❯ make -j 9
❯ ./test/functional/rpc_rawtransaction.py --legacy-wallet
2022-05-07T21:43:41.365000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_7w6fc454
2022-05-07T21:43:41.641000Z TestFramework (ERROR): JSONRPC error
Traceback (most recent call last):
  File "/home/daniela/Developer/bitcoin/test/functional/test_framework/test_framework.py", line 132, in main
    self.setup()
  File "/home/daniela/Developer/bitcoin/test/functional/test_framework/test_framework.py", line 295, in setup
    self.setup_network()
  File "/home/daniela/Developer/bitcoin/./test/functional/rpc_rawtransaction.py", line 72, in setup_network
    super().setup_network()
  File "/home/daniela/Developer/bitcoin/test/functional/test_framework/test_framework.py", line 389, in setup_network
    self.setup_nodes()
  File "/home/daniela/Developer/bitcoin/test/functional/test_framework/test_framework.py", line 416, in setup_nodes
    self.import_deterministic_coinbase_privkeys()
  File "/home/daniela/Developer/bitcoin/test/functional/test_framework/test_framework.py", line 433, in import_deterministic_coinbase_privkeys
    self.init_wallet(node=i)
  File "/home/daniela/Developer/bitcoin/test/functional/test_framework/test_framework.py", line 440, in init_wallet
    n.createwallet(wallet_name=wallet_name, descriptors=self.options.descriptors, load_on_startup=True)
  File "/home/daniela/Developer/bitcoin/test/functional/test_framework/test_node.py", line 753, in createwallet
    return self.__getattr__('createwallet')(wallet_name, disable_private_keys, blank, passphrase, avoid_reuse, descriptors, load_on_startup, external_signer)
  File "/home/daniela/Developer/bitcoin/test/functional/test_framework/coverage.py", line 49, in __call__
    return_val = self.auth_service_proxy_instance.__call__(*args, **kwargs)
  File "/home/daniela/Developer/bitcoin/test/functional/test_framework/authproxy.py", line 144, in __call__
    raise JSONRPCException(response['error'], status)
test_framework.authproxy.JSONRPCException: Compiled without bdb support (required for legacy wallets) (-4)
2022-05-07T21:43:41.694000Z TestFramework (INFO): Stopping nodes
2022-05-07T21:43:41.800000Z TestFramework (WARNING): Not cleaning up dir /tmp/bitcoin_func_test_7w6fc454
2022-05-07T21:43:41.800000Z TestFramework (ERROR): Test failed. Test logging available at /tmp/bitcoin_func_test_7w6fc454/test_framework.log
2022-05-07T21:43:41.801000Z TestFramework (ERROR): 
2022-05-07T21:43:41.801000Z TestFramework (ERROR): Hint: Call /home/daniela/Developer/bitcoin/test/functional/combine_logs.py '/tmp/bitcoin_func_test_7w6fc454' to consolidate all logs
2022-05-07T21:43:41.801000Z TestFramework (ERROR): 
2022-05-07T21:43:41.801000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.
2022-05-07T21:43:41.801000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues
2022-05-07T21:43:41.801000Z TestFramework (ERROR):
```

> I think it might be better to fix this by removing this whole method and also consolidate the two tasks into one:

I've given a bit of thought to this, but I really don't know how I would do that... `self.signrawtransactionwithwallet_tests` still has to use the Core wallet, shouldn't we test it with both the legacy and the descriptor wallet?

Reply:
You can execute the subtests conditionally

```
if sqlite:
  do_sqlite_test()
test()

Reply:
I'm sorry, I really can't figure out how to do this without massively changing the `test_framework.py` code. To me it seems that you can't in a single run test both with the descriptors and the legacy wallet...

Reply:
Right. This should only be possible when both the descriptor wallet and legacy wallet are compiled (and this test is slightly rewritten).

However, I think it is fine to simply run just one (either sqlite or bdb).

This means that any sqlite test case line of code will need to be guarded by a check that checks for sqlite (and the same for bdb). This should not need any framework changes.

Reply:
Unless of course the test is largely monolithic (most lines depend on previous lines), so conditional execution is not straightforward.

Reply:
Let me know if you are stuck on this and I can take a closer look myself.

Reply:
I'm working on it, I'll let you know, thanks 😄

Reply:
Should be ok now :)